assets/js/modules/analytics-4/datastore/audiences.test.js

```diff
@@ -304,6 +307,107 @@ describe( 'modules/analytics-4 audiences', () => {
 			} );
 		} );
 
+		describe( 'maybeSyncAvailableAudiences', () => {
+			it( 'should call syncAvailableAudiences if the availableAudiencesLastSyncedAt setting is undefined', async () => {
+				fetchMock.post( syncAvailableAudiencesEndpoint, {
+					body: availableAudiencesFixture,
+					status: 200,
+				} );
+
+				await registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.maybeSyncAvailableAudiences();
+
+				expect( fetchMock ).toHaveFetchedTimes( 1 );
+				expect( fetchMock ).toHaveFetched(
+					syncAvailableAudiencesEndpoint
+				);
+			} );
+
+			it( 'should not call syncAvailableAudiences if the availableAudiencesLastSyncedAt is within the last hour', async () => {
+				fetchMock.post( syncAvailableAudiencesEndpoint, {
+					body: availableAudiencesFixture,
+					status: 200,
+				} );
+
+				registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetSettings( {
+					availableAudiencesLastSyncedAt:
+						( Date.now() - 1000 ) / 1000, // Value expected to be a PHP date so divide by 1000.
+				} );
+
+				await registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.maybeSyncAvailableAudiences();
+
+				expect( fetchMock ).toHaveFetchedTimes( 0 );
+			} );
+
+			it( 'should call syncAvailableAudiences if the availableAudiencesLastSyncedAt is not within the last hour', async () => {
+				fetchMock.post( syncAvailableAudiencesEndpoint, {
+					body: availableAudiencesFixture,
+					status: 200,
+				} );
+
+				registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetSettings( {
+					availableAudiencesLastSyncedAt:
+						( Date.now() - 2 * 60 * 60 * 1000 ) / 1000, // Value expected to be a PHP date so divide by 1000.
+				} );
+
+				await registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.maybeSyncAvailableAudiences();
+
+				expect( fetchMock ).toHaveFetchedTimes( 1 );
+				expect( fetchMock ).toHaveFetched(
+					syncAvailableAudiencesEndpoint
+				);
+			} );
+
+			it( 'should remove configured audiences which are no longer available', async () => {
+				const availableAudiencesSubset = [
+					availableAudiencesFixture[ 0 ],
+					availableAudiencesFixture[ 2 ],
+				];
+
+				fetchMock.post( syncAvailableAudiencesEndpoint, {
+					body: availableAudiencesSubset,
+					status: 200,
+				} );
+
+				const settings = {
+					configuredAudiences: availableAudiencesFixture.reduce(
+						( acc, a ) => [ ...acc, a.name ],
```

Comment:
Assuming this line is kept what with the requested change to `maybeSyncAvailableAudiences()`:

Please avoid abbreviated variable names as per our naming conventions (I know we do tend to use `acc` so let's let that one slide, but please rename `a` or simply destructure `name`).

```suggestion
						( acc, audience ) => [ ...acc, audience.name ],
```